Mathlib/Analysis/InnerProductSpace/Basic.lean

```diff
@@ -905,6 +905,27 @@ protected theorem Complex.inner (w z : ℂ) : ⟪w, z⟫_ℝ = (conj w * z).re :
 
 end RCLikeToReal
 
+/-- An `RCLike` field is a real inner product space. This is not registered as an instance since it
+creates problems with the case `𝕜 = ℝ`, but it can be used in a proof to obtain a real inner product
+space structure on an `RCLike` field.
+
+This is definitionally different from the inner product space structure obtained from
+`InnerProductSpace.rclikeToReal 𝕜 𝕜` because of a non-defeq diamond in the `ℝ`-normed space
+instances:
+`NormedSpace.restrictScalars ℝ 𝕜 𝕜` isn't defeq to
+`NormedAlgebra.toNormedSpace' (𝕜 := ℝ) (𝕜' := 𝕜)` because the former sets
+`(r : ℝ) • (k : 𝕜) := algebraMap ℝ 𝕜 r * k ` while the latter inherits `r • k` from `RCLike 𝕜`. -/
+noncomputable def RCLike.innerProductSpaceReal : InnerProductSpace ℝ 𝕜 where
```

Comment:
Is this defeq to the obvious instance when `𝕜 = ℝ` or \C ? If so, can you add an `example` to prove it?

Reply:
Both instances are fine:
```
example : (inferInstanceAs <| InnerProductSpace ℝ ℝ) = RCLike.innerProductSpaceReal := rfl
example : (inferInstanceAs <| InnerProductSpace ℝ ℂ) = RCLike.innerProductSpaceReal := rfl
```
I'd argue then that this:
* should be registered as an `instance`
* needs no comment; the diamond is entirely the fault of the other `InnerProductSpace` declaration, not of this one, and so the comment belongs there.

Reply:
There's already a comment on the other `InnerProductSpace` declaration. Do you want me to change it to incorporate some of the comment I was adding to this `InnerProductSpace` declaration?